cvmfs/atomic.h

```diff
@@ -17,92 +17,74 @@ namespace CVMFS_NAMESPACE_GUARD {
   *a = 0;
 }
 
-
 static void inline __attribute__((used)) atomic_init64(atomic_int64 *a) {
   *a = 0;
 }
 
-
 static int32_t inline __attribute__((used)) atomic_read32(atomic_int32 *a) {
   return __sync_fetch_and_add(a, 0);
 }
 
-
 static int64_t inline __attribute__((used)) atomic_read64(atomic_int64 *a) {
   return __sync_fetch_and_add(a, 0);
 }
 
-
-static void inline __attribute__((used)) atomic_write32(
-  atomic_int32  *a,
-  int32_t        value)
-{
-  while (!__sync_bool_compare_and_swap(a, atomic_read32(a), value)) { }
+static void inline __attribute__((used))
+atomic_write32(atomic_int32 *a, int32_t value) {
+  while (!__sync_bool_compare_and_swap(a, atomic_read32(a), value)) {
+  }
 }
 
-
-static void inline __attribute__((used)) atomic_write64(
-  atomic_int64  *a,
-  int64_t       value)
-{
-  while (!__sync_bool_compare_and_swap(a, atomic_read64(a), value)) { }
+static void inline __attribute__((used))
+atomic_write64(atomic_int64 *a, int64_t value) {
+  while (!__sync_bool_compare_and_swap(a, atomic_read64(a), value)) {
+  }
 }
 
-
 static void inline __attribute__((used)) atomic_inc32(atomic_int32 *a) {
-  (void) __sync_fetch_and_add(a, 1);
+  (void)__sync_fetch_and_add(a, 1);
 }
 
-
 static void inline __attribute__((used)) atomic_inc64(atomic_int64 *a) {
-  (void) __sync_fetch_and_add(a, 1);
+  (void)__sync_fetch_and_add(a, 1);
 }
 
-
 static void inline __attribute__((used)) atomic_dec32(atomic_int32 *a) {
-  (void) __sync_fetch_and_sub(a, 1);
+  (void)__sync_fetch_and_sub(a, 1);
 }
 
 static void inline __attribute__((used)) atomic_dec64(atomic_int64 *a) {
-  (void) __sync_fetch_and_sub(a, 1);
+  (void)__sync_fetch_and_sub(a, 1);
 }
 
-static int32_t inline __attribute__((used)) atomic_xadd32(
-  atomic_int32 *a,
-  int32_t offset)
-{
-  if (offset < 0)
-    return __sync_fetch_and_sub(a, -offset);
+static int32_t inline __attribute__((used))
+atomic_xadd32(atomic_int32 *a, int32_t offset) {
+  if (offset < 0) return __sync_fetch_and_sub(a, -offset);
   return __sync_fetch_and_add(a, offset);
 }
 
-
-static int64_t inline __attribute__((used)) atomic_xadd64(
-  atomic_int64 *a,
-  int64_t offset)
-{
-  if (offset < 0)
-    return __sync_fetch_and_sub(a, -offset);
+static int64_t inline __attribute__((used))
+atomic_xadd64(atomic_int64 *a, int64_t offset) {
+  if (offset < 0) return __sync_fetch_and_sub(a, -offset);
   return __sync_fetch_and_add(a, offset);
 }
 
-
-static int32_t inline __attribute__((used)) atomic_cas32(
-  atomic_int32 *a,
-  int32_t cmp,
-  int32_t newval)
-{
-  return __sync_bool_compare_and_swap(a, cmp, newval);
+static int32_t inline __attribute__((used))
+atomic_cas32(atomic_int32 *a, int32_t cmp, int32_t newval) {
+  return __sync_val_compare_and_swap(a, cmp, newval);
 }
 
+static int64_t inline __attribute__((used))
+atomic_cas64(atomic_int64 *a, int64_t cmp, int64_t newval) {
+  return __sync_val_compare_and_swap(a, cmp, newval);
```

Comment:
I think it was no clean. It was calling `__sync_bool_compare_and_swap` but implicitly converting to an int32_t, since this was the return value of the `atomic_cas32` function.

I missed the use of this function in fuse_remount.